pkg/types/vsphere/validation/platform.go

```diff
@@ -74,6 +74,10 @@ func validateVIPs(p *vsphere.Platform, fldPath *field.Path) field.ErrorList {
 		allErrs = append(allErrs, field.Invalid(fldPath.Child("ingressVIP"), p.IngressVIP, err.Error()))
 	}
 
+	if len(p.APIVIP) != 0 && len(p.IngressVIP) != 0 && p.APIVIP == p.IngressVIP {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("IPs"), p.APIVIP, "IPs for both API and Ingress should not be the same."))
```

Comment:
`fldPath.Child` should take the actual json/yaml key. In this case, I think it would be fine to arbitrarily choose one so `IPs` -> `IngressVIP` or `APIVIP`